File: pytools/examples/java/client/protocol/ReqRoleCreate.java
package protocol;


import network.Packet;
import network.PacketUtil;

import java.util.List;
import java.util.ArrayList;


public class ReqRoleCreate
{
	private int uid;
	private int uuid;
	private short sid;
	private short cid;
	private String os;
	private String version;
	private String uname;
	private String source;
	private String source_sub;
	private int login_time;


	public byte[] encode()
	{
		Packet packet = new Packet();
		packet.writeInt(uid);
		packet.writeInt(uuid);
		packet.writeShort(sid);
		packet.writeShort(cid);
		packet.writeString(os);
		packet.writeString(version);
		packet.writeString(uname);
		packet.writeString(source);
		packet.writeString(source_sub);
		packet.writeInt(login_time);
		return packet.encode((short)Msg.P_REQ_ROLE_CREATE);
	}



	public void setUid(int uid)
	{
		this.uid = uid;
	}

	public void setUuid(int uuid)
	{
		this.uuid = uuid;
	}

	public void setSid(short sid)
	{
		this.sid = sid;
	}

	public void setCid(short cid)
	{
		this.cid = cid;
	}

	public void setOs(String os)
	{
		this.os = os;
	}

	public void setVersion(String version)
	{
		this.version = version;
	}

	public void setUname(String uname)
	{
		this.uname = uname;
	}

	public void setSource(String source)
	{
		this.source = source;
	}

	public void setSourceSub(String source_sub)
	{
		this.source_sub = source_sub;
	}

	public void setLoginTime(int login_time)
	{
		this.login_time = login_time;
	}

}

File: pytools/examples/csharp/client/Protocol/ReqRoleCreate.cs
using System.Collections;
using System.Collections.Generic;


public class ReqRoleCreate
{
	private uint _uid;
	private uint _uuid;
	private ushort _sid;
	private ushort _cid;
	private string _os;
	private string _version;
	private string _uname;
	private string _source;
	private string _source_sub;
	private uint _login_time;


	public Packet Encode()
	{
		Packet packet = new Packet();
		packet.WriteUint(this._uid);
		packet.WriteUint(this._uuid);
		packet.WriteUshort(this._sid);
		packet.WriteUshort(this._cid);
		packet.WriteString(this._os);
		packet.WriteString(this._version);
		packet.WriteString(this._uname);
		packet.WriteString(this._source);
		packet.WriteString(this._source_sub);
		packet.WriteUint(this._login_time);
		packet.Encode(Msg.P_REQ_ROLE_CREATE);
		return packet;
	}


	public uint uid
	{
		get { return this._uid; }
		set { this._uid = value; }
	}

	public uint uuid
	{
		get { return this._uuid; }
		set { this._uuid = value; }
	}

	public ushort sid
	{
		get { return this._sid; }
		set { this._sid = value; }
	}

	public ushort cid
	{
		get { return this._cid; }
		set { this._cid = value; }
	}

	public string os
	{
		get { return this._os; }
		set { this._os = value; }
	}

	public string version
	{
		get { return this._version; }
		set { this._version = value; }
	}

	public string uname
	{
		get { return this._uname; }
		set { this._uname = value; }
	}

	public string source
	{
		get { return this._source; }
		set { this._source = value; }
	}

	public string source_sub
	{
		get { return this._source_sub; }
		set { this._source_sub = value; }
	}

	public uint login_time
	{
		get { return this._login_time; }
		set { this._login_time = value; }
	}

}
